examples/howto/data_container.myst.md

```diff
@@ -108,82 +168,177 @@ Coordinates are automatically stored into the {class}`arviz.InferenceData` objec
 
 ```{code-cell} ipython3
-type(data)
+model["observed_temp"].data[:15]
 ```
 
-The methods and functions related to the Data container class are:
+## MutableData
 
-- `data_container.get_value` (method inherited from the theano SharedVariable): gets the value associated with the `data_container`.
-- `data_container.set_value` (method inherited from the theano SharedVariable): sets the value associated with the `data_container`.
-- {func}`pymc.set_data`: PyMC3 function that sets the value associated with each Data container variable indicated in the dictionary `new_data` with it corresponding new value.
+In many cases, you will want the ability to switch out data between model runs. This arises when you want to fit a model to multiple datasets, or if you are interested in out-of-sample prediction. For these cases, `pm.MutableData` is the correct tool.
 
 +++
 
-## Using Data container variables to fit the same model to several datasets
+### Using MutableData container variables to fit the same model to several datasets
 
-This and the next sections are an adaptation of the notebook ["Advanced usage of Theano in PyMC3"](../Advanced_usage_of_Theano_in_PyMC3.html#using-shared-variables) using `pm.Data`.
-
-We can use `Data` container variables in PyMC3 to fit the same model to several datasets without the need to recreate the model each time (which can be time consuming if the number of datasets is large):
+We can use `MutableData` container variables in PyMC to fit the same model to several datasets without the need to recreate the model each time (which can be time consuming if the number of datasets is large):
 
 ```{code-cell} ipython3
-:tags: [hide-output]
-
 # We generate 10 datasets
-true_mu = [rng.random() for _ in range(10)]
+n_models = 10
+true_mu = [rng.random() for _ in range(n_models)]
 observed_data = [mu + rng.random(20) for mu in true_mu]
 
 with pm.Model() as model:
-    data = pm.Data("data", observed_data[0])
+    data = pm.MutableData("data", observed_data[0])
     mu = pm.Normal("mu", 0, 10)
     pm.Normal("y", mu=mu, sigma=1, observed=data)
+```
+
+Once again, the name of our data is `data`, so we can look at it's type. Unlike `pm.ConstantData`, we now see a {meth}`pytensor.compile.sharedvalue.SharedVariable.get_value` of class {class}`pytensor.compile.sharedvalue.SharedVariable` to get the value of the variable. This is because our data is now a `SharedVariable`.
+
+```{code-cell} ipython3
+type(model["data"])
+```
+
+
+The methods and functions related to the Data container class are:
+
+- {meth}`data_container.get_value <pytensor.compile.sharedvalue.SharedVariable.get_value>` (method inherited from the pytensor SharedVariable): gets the value associated with the `data_container`.
+- {meth}`data_container.set_value <pytensor.compile.sharedvalue.SharedVariable.set_value>` (method inherited from the pytensor SharedVariable): sets the value associated with the `data_container`.
```

Comment:
I don't think it's fruitful to discuss these. Just tell users they can call `.eval` to verify which values are stored in the MutableData?

Reply:
I like this better (and it's what I always do myself)